src/mono/wasm/Wasm.Build.Tests/BuildTestBase.cs

```diff
@@ -176,6 +177,21 @@ public BuildTestBase(ProjectProviderBase providerBase, ITestOutputHelper output,
             else if (res.ExitCode == 0)
                 throw new XunitException($"Build should have failed, but it didn't. Process exited with exitCode : {res.ExitCode}");
 
+            // Ensure we got all output.
+            if ((res.ExitCode == 0 && !res.Output.Contains("Build succeeded")) || (res.ExitCode != 0 && !res.Output.Contains("Build failed")))
+            {
+                _testOutput.WriteLine("Replacing dotnet process output with messages from binlog");
+
+                var outputBuilder = new StringBuilder();
+                var buildRoot = BinaryLog.ReadBuild(logFilePath);
+                buildRoot.VisitAllChildren<Message>(m =>
+                {
+                    outputBuilder.AppendLine(m.Title);
+                });
+
+                res = new CommandResult(res.StartInfo, res.ExitCode, outputBuilder.ToString());
```

Comment:
Didn't you overwrite the output that is normally returned by this method? We are failing in 
https://github.com/dotnet/runtime/blob/2c5b609b7c4143d560f700339ca49fe8ab7350b5/src/mono/wasm/Wasm.Build.Tests/Blazor/WorkloadRequiredTests.cs#L171

so it looks like it's looking for a line that should be there but is not. Can't we append to the existing output when we get `res`? And merge that `res.Output` with the previous values from line 174?

Reply:
Normally it would mean that get the output doubled, which might be... weird. My goal (atm) is to override the output only when the process output is incomplete.